packages/core/src/entity/Collection.ts

```diff
@@ -58,10 +58,10 @@ export class Collection<T extends object, O extends object = object> extends Arr
    * The value is cached, use `refresh = true` to force reload it.
    */
   async loadCount(options: LoadCountOptions<T> | boolean = false): Promise<number> {
-    const refresh = typeof options === 'boolean' ? options : typeof options === 'object';
+    const refresh = typeof options === 'boolean' ? options : typeof options === 'object' ? options.refresh ?? false : false;
```

Comment:
please do it the way i proposed, this is very overcomplicated, you dont need 3 ternaries (and nesting ternaries on its own is quite unreadable on its own)

```suggestion
    options = typeof options === 'boolean' ? { refresh: options } : options;
```

and use the `options` directly, no need to destruct anything